General updates

yaman-aljnadi opened the pull request:
## Using this Pull Request Template
1. In the Description of Changes section below, please fill out the details requested
2. Create the pull request
3. Check the boxes in the Type of Change section that apply to your changes
4. Check the boxes in the Checklist Before Review section that apply
5. Assign at least two reviewers to review your pull request by clicking the settings icon next to Reviewers

## Description of Changes
Please document the following about this pull request:
### Identify the organization for which you completed this work
If this work is associated with a university course, please identify the university/course/semester (ex. MTU/MIS3200/Fall2023).
* List the organizaiton here: 

### Identify what team you are on
* List the team here: Yaman Aljnadi

### Identify the task, story, or GitHub issue associated with this work
* Identify the task/story/issue here: Updates

### Identify the particular changes that you made for this pull request
* Describe the changes here: 
1- Added the title option for the documents
2- Added the paragraph limit 
3- Updated the scraping template to be a little bit nicer 
4- moved the delete document function to the management codes

## Type of Change (check all that apply)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation addition/update (non-breaking change that adds documentation to the system)

## Checklist Before Review
- [ ] Do you want to keep your branch for future changes or delete it once it is reviewed?
- [x] I have followed the git practices outlined [here](GitPractices.md) to ensure code conflicts are minimized
- [x] My code follows the style guidelines of this project outlined [here](documentation/StyleGuide.md)
- [ ] Update any dependency files. eg.(requirements.txt)
- [ ] I have performed a self-review of my code. It runs without errors and follows the [Best Coding Practices](BestCodingPractices.md)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] There are no API keys, passwords, usernames, or other secrets in the code. I am using .env files for all secrets.
- [x] I have added tests that prove my fix is effective or that my feature works

yaman-aljnadi committed:
Merge pull request #286 from jeffreywallphd/main

pulling from main

yaman-aljnadi committed:
Updates

1- created the paragraph limit
2- moved the delete document function to management
3- updated the scrape document template

jeffreywallphd reviewed (changes requested):
You are off to a great start. Here are a few notes based on what I'm reviewing so far:
1) I like the simplicity of the new Document Processor page. The choice to select from one of the three tools is much cleaner.
2) On the Document processor, it no longer redirects the user back to the list of documents page. We need to think of this is as a workflow. The user should not have to use the hamburger menu at all when creating a new dataset. The dataset workflow pages should have clear buttons or redirects to move them to the next part of the workflow. The user gets stuck on the Document Processor page. You can either redirect them back to the list of processed documents or have buttons to move them back to the processed documents when they are done scraping multiple documents. I think I like the second option better. Being able to stay on the page and scrape multiple documents would be useful, but a button is needed to keep the user in the workflow.
3) The URL scraper worked fine for me. I didn't get the 500 error your mentioned. I scraped the following site: https://en.wikipedia.org/wiki/Corporate_finance. Please share the link you used for scraping so I can test for a 500 error.
4) On the URL page, is the Output File Name necessary since we have a title now? Are we saving the data to file or database? If to a database, then an an output file might not be necessary anymore.
5) Please remove the scraped output display on the workflow version of the data scraper. I don't want to confuse non-computing users with the HTML output. It should just display the success or failure messages.
6) The Document Details page needs an option to save the dataset to HuggingFace. This option should convert the JSON to the parquet format first. Use the arrow Python library unless you find a better one. Then, the parquet dataset can be pushed to HuggingFace.

yaman-aljnadi committed:
updates

1- Updated the Document Processor page, where the user can upload multiple documents. Also added a button to direct the user to process the added data

2- Updated the fiels delete functionality

yaman-aljnadi committed:
Adding the ability to push to hugging face

jeffreywallphd reviewed (changes requested):
These changes are excellent. I think we are almost there. Here are a couple of small changes to make and then I think the workflow is ready:
1) Let's rename the "Upload Parquet" button to "Save to HuggingFace"
2) We need a textbox associated with the HuggingFace save button that allows the user to choose a name for the dataset. The box should be clearly labeled as the save name for the dataset.
3) The parquet file saves to HuggingFace without problem. However, the parquet file on HuggingFace has errors. On first glance, I believe it is a problem with the JSON structure. The "questions" lists and "part" key are likely to blame. You may need a flatter data structure than you currently have. You can view the full error message on HuggingFace by clicking on the saved dataset and then clicking on the text "The dataset viewer is not available for this split."

yaman-aljnadi committed:
Update Names, and adding the custom name field

yaman-aljnadi committed:
Updating the uploading to parquet, to handle the normalization

yaman-aljnadi committed:
Adding the CSV option

yaman-aljnadi committed:
Custom Dataset Name

yaman-aljnadi committed:
Updating the prompt to pop-up

yaman-aljnadi committed:
legal filter

jeffreywallphd committed:
save to Huggingface using Dataset

yaman-aljnadi committed:
Updating_The_Settings_and_adding_multi_database_select

jeffreywallphd committed:
Merge branch 'main' into SPRING2025-GEN-UPDATES

jeffreywallphd committed:
updated upload to HuggingFace to push dataset name correctly

jeffreywallphd reviewed (approved):
I liked your addition of the HuggingFace org lookup. With the changes you created, the upload to HuggingFace was still limited to existing datasets. I changed it to pull the orgs instead of the datasets in the upload dialog. I also utilized the HF Dataset module instead of the file upload method. The dataset now uploads with the name provided in the dialog. I also added the capability to include an instruction prompt. Everything seems to be working great.
